Tests/Opc.Ua.Gds.Tests/CertificateGroupTests.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Opc.Ua.Gds.Server;
+
+namespace Opc.Ua.Gds.Tests
+{
+    [TestFixture, Category("GDS")]
+    [SetCulture("en-us"), SetUICulture("en-us")]
+    [Parallelizable]
+    public class CertificateGroupTests
+    {
+        #region Test Methods
+
+        [Test]
+        public void TestCreateCACertificateAsyncThrowsException()
+        {
+            var configuration = new CertificateGroupConfiguration();
+            configuration.SubjectName = "CN=GDS Test CA, O=OPC Foundation";
+            string path = Utils.ReplaceSpecialFolderNames("%LocalApplicationData%/OPC/GDS/Teststore");
+            var certificateGroup = new CertificateGroup().Create(path, configuration);
+            Assert.That(() => certificateGroup.CreateCACertificateAsync("This is not the ValidSubjectName for my CertificateGroup"), Throws.TypeOf<ArgumentException>());
+        }
+
+        [Test]
+        public async Task TestCreateCACertificateAsyncCertIsInTrustedStoreAsync()
+        {
+            var configuration = new CertificateGroupConfiguration();
+            configuration.SubjectName = "CN=GDS Test CA, O=OPC Foundation";
+            string path = Utils.ReplaceSpecialFolderNames("%LocalApplicationData%/OPC/GDS/Teststore");
+            var certificateGroup = new CertificateGroup().Create(path, configuration);
+            var certificate = await certificateGroup.CreateCACertificateAsync(configuration.SubjectName).ConfigureAwait(false);
+            Assert.NotNull(certificate);
+            using (ICertificateStore trustedStore = CertificateStoreIdentifier.OpenStore(configuration.TrustedListPath))
+            {
+                X509Certificate2Collection certs = await trustedStore.FindByThumbprint(certificate.Thumbprint).ConfigureAwait(false);
+                Assert.IsTrue(certs.Count == 1);
+            }
+            Directory.Delete(path, true);
```

Comment:
if any of the asserts fails, this will not get executed, leaving a stale folder on the disk.